payments-contracts/src/test/kotlin/com/gevamu/corda/test/contracts/PaymentContractUnknownStateTypeTest.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2023 Exactpro Systems Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.gevamu.corda.test.contracts
+
+import com.gevamu.corda.contracts.PaymentContract
+import com.gevamu.corda.states.Payment
+import net.corda.core.contracts.BelongsToContract
+import net.corda.core.contracts.ContractState
+import net.corda.core.identity.AbstractParty
+import net.corda.core.identity.Party
+import net.corda.core.node.services.AttachmentId
+import net.corda.testing.node.ledger
+import org.junit.jupiter.api.Test
+import java.time.Instant
+import java.util.UUID
+
+class PaymentContractUnknownStateTypeTest : AbstractPaymentContractTest() {
+    @BelongsToContract(PaymentContract::class)
+    data class Dummy(
+        val payer: Party,
+        val gateway: Party,
+        val endToEndId: String,
+        val paymentInstructionId: AttachmentId,
+        val status: Payment.PaymentStatus,
+        val additionalInfo: String? = null,
+        val uniquePaymentId: UUID = UUID.randomUUID(),
+        val timestamp: Instant = Instant.now()
+    ) : ContractState {
+        override val participants: List<AbstractParty>
+            get() = if (status == Payment.PaymentStatus.CREATED) listOf(payer) else listOf(payer, gateway)
+    }
+
+    class DummyCommand(override val uniquePaymentId: UUID) : PaymentContract.Commands
+
+    @Test
+    fun `test ignore different type of state`() {
+        val dummy = Dummy(
+            uniquePaymentId = uniquePaymentId,
+            payer = payer.party,
+            gateway = gateway.party,
+            endToEndId = endToEndId,
+            paymentInstructionId = attachmentId,
+            status = Payment.PaymentStatus.CREATED
+        )
+        ledgerServices.ledger {
+            transaction {
+                command(payer.publicKey, PaymentContract.Commands.Create(uniquePaymentId))
+                output(PaymentContract.ID, dummy)
+                verifies()
```

Comment:
This test shouldn't actually pass. Looks like we need to also check for commands which don't have corresponding input/output states.